### ISSUE TITLE
@AfterViews being called three times

### ISSUE THREAD

**Issue by jordansilva** [androidannotations/androidannotations#449]
Bug?
Methods with @AfterViews began to be called 3 times when inherited from class SlidingFragmentActivity (SlidingMenu library). The OnCreate is called only once.

**Comment by mc-android-developer** [comment]
It's not only SlidingActivity issue. I have the same problem in some other cases and I don't use SlidingActivity.

As it was critical for me I investigate source and found that a problem in org.androidannotations.api.view.OnViewChangedNotifier class.
Speaking shortly, I have a single instance of fragment in my app. The fragment show some information, so I attach and remove it from activity from time to time.

Any time I attach the Fragment method OnViewChangedNotifier#registerOnViewChangedListener called and my fragment added to OnViewChangedNotifier#listeners
As 'listeners' is a List and there are no extra checks inside registerOnViewChangedListener my fragment(the same instance) added there a few times.

 After that OnViewChangedNotifier#notifyViewChanged called in which 'listeners' are iterated and OnViewChangedListener#onViewChanged method in my fragment called as many times as it presents in the list.

I guess the simplest solution would be to change List to Set for OnViewChangedNotifier#listeners. I did   that first, but then I noticed it had already done in source code on GitHub. However, I tried this on the very last 3.2 version and problem is still there. I guess it will disappear in the next version.
If it is critical for you, compile lib from the last sources from GitHub and it will work fine.

**Comment by WonderCsabo** [comment]
@mc-android-developer you do not have to compile to library, since we push a [snapshot release](https://oss.sonatype.org/content/repositories/snapshots/org/androidannotations/) after each commit to the `develop` branch.

**Comment by sbrough** [comment]
I'm seeing this in a plain Activity with @AfterViews getting called twice.

I've also got a BaseActivity that is extending the ActionBarActivity, and any activity that inherits from that BaseActivity is calling is calling @AfterViews three times.

Any suggestions?

**Comment by WonderCsabo** [comment]
Try to debug the code (put a breakpoint) to see what calls your `@AfterViews`  methods actually.
